src/posts/introducing-pint-xarray/index.md

```diff
@@ -0,0 +1,295 @@
+
+```python
+time = xr.DataArray([0.1, 0.2, 0.3], dims='time')
+distance = xr.DataArray(name='distance',
+                        data=[10, 20, 25],
+                        dims=['time'],
+                        coords={'time': time})
+distance = distance.pint.quantify({'distance': 'metres',
+                                   'time': 'seconds'})
+print(distance.coords['time'].attrs)
+```
+
+```
+Out: {'units': <Unit('second')>}
+```
+
+This allows us to provide conveniently wrapped versions of common xarray methods like `.sel`, so that you can still select subsets of data in a unit-aware fashion like this:
+
+```python
+distance.pint.sel(time=200 * pint.Unit('milliseconds'))
+```
+
+```
+Out:
+<xarray.DataArray 'distance' ()>
+<Quantity(20, 'meter')>
+Coordinates:
+    time     float64 200.0
+```
+
+Observe how the `.pint.sel` operation has first converted 200 milliseconds to 0.2 seconds, before finding the distance value that occurs at a time position of 0.2 seconds.
+
+[This wrapping is currently necessary](https://xarray.pydata.org/en/stable/user-guide/duckarrays.html#missing-features) for any operation which needs to be aware of the units of a dimension coordinate of the dataarray, or any xarray operation which relies on an external library (such as calling `scipy` in `.integrate`).
+
+## CF-compliant units for geosciences with cf-xarray
+
+Different fields tend to have different niche conventions about how certain units are defined.
+By default, pint doesn't understand all the unusual units and conventions we use in geosciences.
+But [pint is customisable](https://pint.readthedocs.io/en/stable/defining.html), and with the help of [cf-xarray](https://github.com/xarray-contrib/cf-xarray) we can teach it about these geoscience-specific units.
+
+If we `import cf_xarray.units` (before `import pint_xarray`) then we can `quantify` example data from a pangeo climate data store :
+
+```python
+import xarray as xr
+import cf_xarray.units  # must come before import pint_xarray
+import pint_xarray
+
+ds = xr.open_dataset('gs://cmip6/CMIP6/CMIP/NCAR/CESM2-FV2/historical/r2i1p1f1/Amon/sfcWind/gn/v20200226/', engine='zarr')
+ds = ds.pint.quantify()
+
+squared_wind = ds['sfcWind'] ** 2
+squared_wind.pint.units
+```
+
+```
+Out: <Unit('meter ** 2 / second ** 2')>
+```
+
+Here (thanks to `cf_xarray`) pint has successfully interpreted the CF-style units `'m s-1'`, then automatically changed them when we squared the wind speed.
+
+## Plotting
+
+We can complete our real-world example by plotting the data in its new units
+
+```python
+import cartopy.crs as ccrs
+import matplotlib.pyplot as plt
+
+p = squared_wind.isel(time="2014-01").plot(
+    subplot_kws=dict(projection=ccrs.Orthographic(-80, 35), facecolor="gray"),
+    transform=ccrs.PlateCarree(),
+)
+p.axes.set_global()
+p.axes.coastlines()
+plt.show()
+```
+
+![cartopy plot of a quantified dataset](squared_wind.png)
```

Comment:
one more change :) 

```suggestion
![cartopy plot of a quantified dataset](/posts/introducing-pint-xarray/squared_wind.png)
```

Reply:
thanks, I was about to try looking up what we did while testing the blog section